### PR TITLE
Migrate docs: Setup VAST

### PR DESCRIPTION
**Pull request by mavam**
This PR migrates and updates the installation instructions.

### :dart: Review Instructions

Read the new text entirely. I've made a lot of modifications.

### CODE REVIEW

File: docs/docs/setup-vast/install/linux.md
Comment (mavam) [inline docs/docs/setup-vast/install/linux.md:103]: @satta I've lost track about the current issues we face regarding Debian packaging. Mind providing a brief update?

File: docs/docs/setup-vast/install/macos.md
Comment (mavam) [inline docs/docs/setup-vast/install/macos.md:1]: @dominiklohmann should we add Homebrew instructions here as well?

File: docs/docs/setup-vast/install/linux.md
Comment (mavam) [inline docs/docs/setup-vast/install/linux.md:74]: This feels a bit too specific, and since the CLI is changing in the future, we probably should rewrite this.

File: docs/docs/setup-vast/install/docker.md
Comment (mavam) [inline docs/docs/setup-vast/install/docker.md:20]: This feels a bit excessive. Isn't there a shorter way?

File: docs/docs/setup-vast/install/docker.md
Comment (mavam) [inline docs/docs/setup-vast/install/docker.md:28]: @dominiklohmann is `TAG` optional?

File: docs/docs/setup-vast/install/macos.md
Comment (dominiklohmann) [reply docs/docs/setup-vast/install/macos.md:1]: That no longer exists. It's just building from source now. Installing via CMake also installs a launchd config file that can be enabled via `launchctl load -w ~/Library/LaunchAgents/com.tenzir.vast.plist`, but that's entirely optional.

File: docs/docs/setup-vast/install/docker.md
Comment (dominiklohmann) [reply docs/docs/setup-vast/install/docker.md:20]: Not really, no.

- `-d` for detaching, i.e., running in background
- `-t` for terminal output
- `--name` to name the image
- `--rm` to clear older volumes
- `-p` to expose the port to the outer world
- `-v` to mount the database directory

File: docs/docs/setup-vast/install/docker.md
Comment (dominiklohmann) [reply docs/docs/setup-vast/install/docker.md:28]: It defaults to `latest`. We upload this image though, not sure why we're providing build instructions here. The user should only be interested in _pulling_ the image.

File: docs/docs/setup-vast/install/linux.md
Comment (dominiklohmann) [reply docs/docs/setup-vast/install/linux.md:8]: If `CPack` is available on the system you're building on you automatically get a `package` build target that packages VAST. Because this does not currently include systemd/launchd/rc.d configuration I don't think it's worth mentioning here (yet).

File: docs/docs/setup-vast/install/macos.md
Comment (mavam) [reply docs/docs/setup-vast/install/macos.md:1]: Can you provide me snippet/diff that I can apply to this branch?

Comment (dominiklohmann) [reply docs/docs/setup-vast/install/macos.md:1]: I'd just append one paragraph at the bottom:

> ## launchd
> 
> Installing VAST via CMake on macOS installs a [launchd agent](https://www.launchd.info) at `~/Library/LaunchAgents/com.tenzir.vast.plist`. To run VAST automatically on login, run this command:
> ```bash
> # NOTE: Use 'unload' rather than 'load' to unload the agent
> launchctl load -w ~/Library/LaunchAgents/com.tenzir.vast.plist
> ```

File: docs/docs/setup-vast/install/docker.md
Comment (mavam) [reply docs/docs/setup-vast/install/docker.md:20]: Thanks, I added this explanation to the instructions.

File: docs/docs/setup-vast/install/linux.md
Comment (mavam) [reply docs/docs/setup-vast/install/linux.md]: Okay.

File: docs/docs/setup-vast/install/linux.md
Comment (satta) [reply docs/docs/setup-vast/install/linux.md:101]: Yes. From my point of view I am currently seeing the following roadblocks for providing up-to-date packages for VAST:

* Lack of packaging for VAST dependencies in the required versions, most importantly Apache Arrow. See this [Request For Package (ITP)](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=970021). A preliminary package [exists](https://salsa.debian.org/science-team/arrow) but needs stable maintainers who are willing and able to follow upstream's fast development process and ensure that new breaking releases are dealt with in a Debian policy-compliant way.
* (Minor issue) Lack of support for some (most?) non-Intel architectures that Debian supports. Not a dealbreaker but a nice-to-have.
* Missing ABI stability in VAST to provide a proper shared library package plus development headers package. Each ABI break would mean a new ABI version bump -- this essentially means that each ABI change would cause the new package to be treated like a completely new and unreviewed package that has to be manually cleared. Moreover one would need to ensure that the VAST version always matches the libvast version since IIRC there is no backward compatibility.

Comment (dominiklohmann) [reply docs/docs/setup-vast/install/linux.md:101]: The third bullet would essentially require us to define a stable subset of libvast that we offer as public API with a stable ABI, but that adds impediments that no sane startup wants to deal with if they don't need to. I don't think there'll be an agreement between us and the Debian way of doing things in this regard for years to come.

We should probably just erase the section here from the docs page.

File: docs/docs/setup-vast/install/linux.md
Comment (dominiklohmann) [reply docs/docs/setup-vast/install/linux.md:72]: Another way to set this up is to set `vast.start.commmands: [import pcap]`, which is a lot more intuitive and probably what we should be recommending instead.

Thinking of it further, it'd actually be nice to have some `vast remote-exec` command for simply executing a command in the context of a remote VAST instance. That could supersede `spawn source` a lot more generically, and would probably only take a few hours to gobble up. :thinking:

File: docs/docs/setup-vast/install/macos.md
Comment (mavam) [reply docs/docs/setup-vast/install/macos.md]: I s/copies/configures/ for you. 🌻

File: docs/docs/setup-vast/install/linux.md
Comment (mavam) [reply docs/docs/setup-vast/install/linux.md:72]: > Another way to set this up is to set `vast.start.commmands: [import pcap]`, which is a lot more intuitive and probably what we should be recommending instead.

Yes, that's also possible. My point was rather to not mention start commands at all, because my hunch is they will be much less relevant after the Event Router.

> Thinking of it further, it'd actually be nice to have some `vast remote-exec` command for simply executing a command in the context of a remote VAST instance.

That's probably the no-longer-existent `vast send` command, which basically allowed for sending messages to component actors from via the CLI.

File: docs/docs/setup-vast/install/docker.md
Comment (mavam) [reply docs/docs/setup-vast/install/docker.md]: I've revamped the Docker docs. Mind giving it a read?

File: docs/docs/setup-vast/install/linux.md
Comment (mavam) [reply docs/docs/setup-vast/install/linux.md:101]: Thanks for laying it out, @satta. I think (1) and (2) are possible to overcome.

(3) is complicated. ABI stability for libvast is not feasible for us, as @dominiklohmann pointed out. 

That said, our *plugin API* is a stable point of reference. I'm not sure if Debian requires true ABI stability or whether a subset suffices. In any case, our main dependency Arrow is also moving fast, without ABI guarantees, so the problem also applies to (1).

If ABI stability is not negotiable, we probably have to create our own repository where we simply serve DEBs for libvast, heeaders, and plugins. That said, before considering "our own", we probably would try to find out if [OBS](https://openbuildservice.org/) suits the bill.

File: docs/docs/setup-vast/install/linux.md
Comment (dominiklohmann) [reply docs/docs/setup-vast/install/linux.md:72]: It's actually slightly different from the `vast send` command in that it does not send a message to an actor, but rather only communicates via commands with the node, which is much safer. I reckon this would be a nice hackathon project, maybe for our next one. :)

Anyways, we can also erase the section. I'd rather not document it than mention `vast spawn`, because that command has absolutely no test coverage.

File: docs/docs/get-started/README.md
Comment (dominiklohmann) [inline docs/docs/get-started/README.md:29]:
```suggestion
  click use "/vast/docs/use-vast/" "Use VAST"
  click learn "/vast/docs/learn-vast/" "Learn VAST"
  click develop "/vast/docs/develop-vast/" "Develop VAST"
```

Comment (mavam) [reply docs/docs/get-started/README.md:29]: Jeez, your eyes!

File: docs/docs/setup-vast/tune.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/tune.md:62]: This should probably also mention `active-partition-timeout`, which can override this upper bound.

File: docs/docs/setup-vast/tune.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/tune.md:219]: This default only applies to the server process. For client processes, it is 100.

File: docs/docs/setup-vast/tune.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/tune.md:179]: Is there a way to render these calculations nicer?

Also, it probably should be a capital B for Bytes rather than b for bits throughout the calculations.

I also wonder whether these calculations are still correct given that we now have multiple active partitions (@tobim?).

File: docs/docs/setup-vast/tune.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/tune.md:127]: I find that this section distracts from tuning. It's much too complicated for most readers, so maybe it should go into some separate section with advanced tutorials or so. It also appears to be out of date already.

File: docs/docs/setup-vast/tune.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/tune.md:135]: Likely outdated as this still refers to the archive segments.

File: docs/docs/setup-vast/download.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/download.md:13]:
```suggestion
Get the source code by cloning our Git repository or downloading an archive.

### Git
```

The official spelling is with an uppercase G.

File: docs/docs/setup-vast/download.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/download.md:48]:
```suggestion
For development versions or a specific Git commit, you need to navigate through
the GitHub CI to find the build artifact:
```

File: docs/docs/setup-vast/download.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/download.md:56]: I don't think this section adds much value, and would rather see it removed. It complicates an otherwise straightforward section.

File: docs/docs/setup-vast/download.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/download.md:65]:
```suggestion
[Docker Hub](https://hub.docker.com/repository/docker/tenzir/vast).
```

Official spelling uses two words.

File: docs/docs/setup-vast/install/linux.md
Comment (dominiklohmann) [inline docs/docs/setup-vast/install/linux.md:46]: Why not just recommend using the default directory, e.g., `$PREFIX/etc/vast/vast.yaml`? Then it wouldn't need to be specified here. We also don't specify it in the example in the repository.

**Review by dominiklohmann — APPROVED** [review]
Just some minor requests, this looks really good. Approving already assuming you agree with all my comments.

File: docs/docs/setup-vast/tune.md
Comment (mavam) [reply docs/docs/setup-vast/tune.md]: I've removed it. We'll bring it back after it's fixed. (I created an internal story for it.)

File: docs/docs/setup-vast/download.md
Comment (mavam) [reply docs/docs/setup-vast/download.md:56]: Disagreeing on this one, I needed this more than once in the past.

Comment (dominiklohmann) [reply docs/docs/setup-vast/download.md:56]: But you're a developer, not a user.

File: docs/docs/setup-vast/install/linux.md
Comment (mavam) [reply docs/docs/setup-vast/install/linux.md:46]: I don't know. @lava or @tobim might know.

(The content is unadapted from the existing docs.)

File: docs/docs/setup-vast/tune.md
Comment (mavam) [reply docs/docs/setup-vast/tune.md]: Good call.

Comment (mavam) [reply docs/docs/setup-vast/tune.md]: Added a sentence.

File: docs/docs/setup-vast/install/linux.md
Comment (dominiklohmann) [reply docs/docs/setup-vast/install/linux.md:46]: Note that duplicating the config from `$PREFIX/etc/vast/vast.yaml` with `--config` is not side-effect free per sc-32665; we should definitely not be recommending doing so, and if you want to see the used config files that already is available in `vast status | jq .system`.

File: docs/docs/setup-vast/download.md
Comment (mavam) [reply docs/docs/setup-vast/download.md]: This is not about developers. This is about the user that wants to be running the bleeding-edge version.

Comment (mavam) [reply docs/docs/setup-vast/download.md]: It's _really_ handy to throw someone a static build, especially after we pushed a fix on master.

Comment (mavam) [reply docs/docs/setup-vast/download.md]: But we're not recommending it. We're giving the user the option. Nobody goes through these hoops unless they have a specific reason.

File: docs/docs/setup-vast/install/linux.md
Comment (mavam) [reply docs/docs/setup-vast/install/linux.md]: I removed the contested `--config` option. If there is a strong opinion to keep it, I would create another internal story.

**Pull request merged**
